### PR TITLE
[ZEPPELIN-5600] Support Flink 1.15

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This PR is to support Flink 1.15 in Zeppelin.
* One module `flink1.15-shims` is introduced. Most of the flink 1.15 specific code is in this module.
* Only scala 2.12 is supported for Flink 1.15
* Flink 1.15 has changed its binary distribution, user need to do extra configuration(See `flink.md`)

### What type of PR is it?
[Feature ]

### Todos
* [ ] - Task

### What is the Jira issue?
*https://issues.apache.org/jira/browse/ZEPPELIN-5600

### How should this be tested?
* CI is passed, and manually tested flink 1.15

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: flink/flink-scala-parent/pom.xml [hunk 1]
```diff
@@ -41,10 +41,19 @@
...
     <flink.bin.download.url>https://archive.apache.org/dist/flink/flink-${flink.version}/flink-${flink.version}-bin-scala_${flink.scala.binary.version}.tgz</flink.bin.download.url>
   </properties>
 
+  <repositories>
+    <repository>
+      <id>flink-115-repo</id>
+      <name>flink-115-repo</name>
+      <url>https://repository.apache.org/content/repositories/orgapacheflink-1488/</url>
+    </repository>
+  </repositories>
```
Comment (kbendick) [inline flink/flink-scala-parent/pom.xml:56]: Flink 1.15 has been released, so all of the staging repositories for the release candidates have been taken down per ASF policy.

Note the 1.15.0 jar for `flink-streaming-java` on maven central: https://mvnrepository.com/artifact/org.apache.flink/flink-streaming-java/1.15.0 🙂

Comment (zjffdu) [reply flink/flink-scala-parent/pom.xml:56]: Thanks for review @kbendick Let me fix it.

Comment (kbendick) [reply flink/flink-scala-parent/pom.xml]: No problem. Hope I'm not intruding. I work in the Apache Iceberg community mostly, and have been using some beta builds of this PR for Flink 1.15 testing.

Comment (zjffdu) [reply flink/flink-scala-parent/pom.xml]: That's awesome, I am creating an iceberg tutorial in Zeppelin notebook, hope that will be helpful for iceberg community

**Pull request merged**